t/t4216-log-bloom.sh

```diff
@@ -0,0 +1,99 @@
+#!/bin/sh
+
+test_description='git log for a path with bloom filters'
+. ./test-lib.sh
+
+test_expect_success 'setup test - repo, commits, commit graph, log outputs' '
+	git init &&
+	mkdir A A/B A/B/C && 
+	test_commit c1 A/file1 &&
+	test_commit c2 A/B/file2 &&
+	test_commit c3 A/B/C/file3 &&
+	test_commit c4 A/file1 &&
+	test_commit c5 A/B/file2 &&
+	test_commit c6 A/B/C/file3 &&
+	test_commit c7 A/file1 &&
+	test_commit c8 A/B/file2 &&
+	test_commit c9 A/B/C/file3 &&
+	git checkout -b side HEAD~4 &&
+	test_commit side-1 file4 &&
+	git checkout master &&
+	git merge side &&
+	test_commit c10 file5 &&
+	mv file5 file5_renamed &&
+	git add file5_renamed && 
+	git commit -m "rename" &&
+	git commit-graph write --reachable --changed-paths
+'
+
+setup() {
+	rm output 
+	rm "$TRASH_DIRECTORY/trace.perf"
+	git -c core.commitGraph=false log --pretty="format:%s" $1 >log_wo_bloom
+	GIT_TRACE2_PERF="$TRASH_DIRECTORY/trace.perf" git -c core.commitGraph=true log --pretty="format:%s" $1 >log_w_bloom
+}
+
+test_bloom_filters_used() {
+	local log_args=$1
+	local bloom_trace_prefix="statistics:{\"filter_not_present\":0,\"zero_length_filter\":0,\"maybe\""
+	setup "$log_args"
+	grep -q "$bloom_trace_prefix" "$TRASH_DIRECTORY/trace.perf" && test_cmp log_wo_bloom log_w_bloom
+}
+
+test_bloom_filters_not_used() {
+	local log_args=$1
+	setup "$log_args"
+	!(grep -q "statistics:{\"filter_not_present\":" "$TRASH_DIRECTORY/trace.perf") && test_cmp log_wo_bloom log_w_bloom
+}
+
+test_log_options_all_simple_paths() {
+	local result=0
+	for path in A A/B A/B/C A/file1 A/B/file2 A/B/C/file3 file4 file5_renamed
+	do
+		for option in "" "--full-history" "--full-history --simplify-merges" "--simplify-merges" "--simplify-by-decoration" \
```

Comment:
Perhaps break this line up more using `\` to say we continue to the next line?

```
for option in "" \
              "--full-history" \
              "--full-history --simplify-merges" \
              "--simplify-merges" \
              "--simplify-by-decoration" \
              "--follow" \
              "--first-parent" \
              "--topo-order" \
              "--date-order" \
              "--author-date-order"
```

Also, I would drop the `--*-order` operations, as those just change commit order, they do not change what is "interesting".  Maybe make sure that one of your commits performs an exact file rename so that `--follow` is tested correctly.

Reply:
(I see you have a renamed file now)

Reply:
@derrickstolee The reason I added the order tests is because 
a) I want to clearly document all the options that are using and benefiting from bloom filters and more importantly,
b) those options were broken without the fix I made to move the prepare_to_use_bloom_filters call into check_maybe_different* method